Tasks/JenkinsQueueJob/job.ts

```diff
@@ -308,6 +320,62 @@ export class Job {
             }).auth(thisJob.queue.taskOptions.username, thisJob.queue.taskOptions.password, true);
         }
     }
+
+    downloadResults(): void {
+        var thisJob: Job = this;
+        var downloadUrl: string = Util.addUrlSegment(thisJob.executableUrl, 'team-results/zip');
+        tl.debug('downloadResults(), url:' + downloadUrl);
+
+        var downloadRequest = request.get({ url: downloadUrl, strictSSL: thisJob.queue.taskOptions.strictSSL })
+            .auth(thisJob.queue.taskOptions.username, thisJob.queue.taskOptions.password, true)
+            .on("error", err => {
+                Util.handleConnectionResetError(err); // something went bad
+                thisJob.stopWork(thisJob.queue.taskOptions.pollIntervalMillis, thisJob.state);
+            })
+            .on("response", response => {
+                tl.debug('downloadResults(), response.statusCode: ' + response.statusCode);
+                if (response.statusCode != 200) {
+                    tl.debug('no results to download');
+                    thisJob.stopWork(0, JobState.Done);
+                } else { // 200
+                    var destinationFolder: string = path.join(thisJob.queue.taskOptions.saveResultsTo, thisJob.name + '/')
+                    var fileName = path.join(destinationFolder, 'team-results.zip');
+
+                    try {
+                        // Create the destination folder if it doesn't exist
+                        if (!tl.exist(destinationFolder)) {
+                            tl.debug('creating results destination folder: ' + destinationFolder);
+                            tl.mkdirP(destinationFolder);
+                        }
```

Comment:
The expectation is that it will not exist as the staging directory is cleared prior to each build.  There could be a collision if a pipeline triggered multiple of the same jobs though.